### ISSUE TITLE
Allow edit recurrent contribution

### ISSUE THREAD

**Issue by francescbassas** [Project60/org.project60.sepa#706]
We are starting to use the extension for the first time.

If I am not mistaken, once a mandate has been generated, there is no option to modify the recurring contribution to be able to modify, for example, the amount to be direct debited. Is it for some reason or is it simply that this functionality has not been enabled?

**Comment by Detsieber** [comment]
From a legal point of view, it would be no problem to have differing amounts for ongoing mandates. So, no reason why we shouldn't have that in CiviSEPA.

However: It is not too long ago, that changing recurring contributions was enabled in CiviCRM core. So there is probably some work to do for enhancing CiviSEPA. And this probably would need funding...

**Comment by francescbassas** [comment]
Thanks @Detsieber.

Stripe extensions allow it, can serve as a reference. Taking a quick look I think [this variable](https://docs.civicrm.org/dev/en/latest/extensions/payment-processors/paymentclass/#supportseditrecurringcontribution) needs to be defined.

We will assess the need and perhaps propose some PR. However, we have generated a report with SearchKit that also allows us to edit the amounts of recurring contributions and will surely serve as a workaround.

**Comment by francescbassas** [comment]
@bjendres if I understood you correctly... mandates of type `FLEX`  would allow you to edit their amounts and installments without the necessity of regenerate the mandate. This would adjust to the need to be able to modify the amounts owed, for example when we modify the price of the subscribed magazine.

**Comment by bjendres** [comment]
> .. mandates of type `FLEX` would allow you to edit their amounts and installments without the necessity of regenerate the mandate. 

Not quite. Mandates of type ``FLEX`` would call a hook/api/function during the batching process to return the next contribution (if any) to be collected on this mandate.

However, this is just a concept so far, and it's not clear if the organisation we created it for is going to go for it.

**Comment by bjendres** [comment]
Yes, I assumed you had seen that already. Does that work for you?

**Comment by GFF-DS** [comment]
> I think so. I'm still getting used to using the extension.

@francescbassas Did you experience any problems since your last post? Can you confirm that in your experience changing the amount for a recurring SEPA mandate works well?
